components/lock/redis/standalone_redis_lock.go

```diff
@@ -0,0 +1,198 @@
+package redis
+
+import (
+	"context"
+	"crypto/tls"
+	"errors"
+	"fmt"
+	"github.com/go-redis/redis/v8"
+	"mosn.io/layotto/components/lock"
+	"mosn.io/pkg/log"
+	"strconv"
+	"time"
+)
+
+const (
+	host                   = "redisHost"
+	password               = "redisPassword"
+	enableTLS              = "enableTLS"
+	maxRetries             = "maxRetries"
+	maxRetryBackoff        = "maxRetryBackoff"
+	defaultBase            = 10
+	defaultBitSize         = 0
+	defaultDB              = 0
+	defaultMaxRetries      = 3
+	defaultMaxRetryBackoff = time.Second * 2
+	defaultEnableTLS       = false
+)
+
+// Standalone Redis lock store.Any fail-over related features are not supported,such as Sentinel and Redis Cluster.
+type StandaloneRedisLock struct {
+	client   *redis.Client
+	metadata metadata
+	replicas int
+
+	features []lock.Feature
+	logger   log.ErrorLogger
+
+	ctx    context.Context
+	cancel context.CancelFunc
+}
+
+// NewStandaloneRedisLock returns a new redis lock store
+func NewStandaloneRedisLock(logger log.ErrorLogger) *StandaloneRedisLock {
```

Comment:
I think we also need to add some tests for the component implementations.

Reply:
fixed.I added a ut for this component,which starts a mini-redis for test